app/models/observation_accuracy_experiment.rb

```diff
@@ -54,50 +53,62 @@ def fetch_top_iders( window: nil, taxon_id: nil, size: nil, place_id: nil )
     iders
   end
 
-  def get_improving_identifiers( candidate_iders: nil, taxon_id: nil, place_id: nil, window: nil, page: 1 )
-    params = {
-      category: "improving",
-      per_page: 200,
-      page: page,
-      user_id: candidate_iders,
-      d1: window,
-      taxon_id: taxon_id,
-      place_id: place_id
-    }.compact
-    ids = INatAPIService.get( "/identifications", params )
-    ids&.results&.map {| r | r.dig( "user", "id" ) } || []
-  end
-
-  def get_qualified_candidates( taxon_id: nil, window: nil, top_iders_only: true, place_id: nil )
-    if top_iders_only
-      candidate_top_taxon_iders = fetch_top_iders( window: window, taxon_id: taxon_id, place_id: place_id ).
-        select {| _, v | v >= improving_id_threshold }.keys
-      return [] if candidate_top_taxon_iders.empty?
-
-    else
-      candidate_top_taxon_iders = nil
-    end
-    improving_identifiers = []
-    enough_candidates = false
-    page = 1
-    sorted_frequency_hash = {}
-    until enough_candidates
-      new_values = get_improving_identifiers(
-        candidate_iders: candidate_top_taxon_iders,
-        taxon_id: taxon_id,
-        place_id: place_id,
-        window: window,
-        page: page
+  def get_improving_identifiers( taxon_id: nil, place_id: nil, window: nil )
+    filter = [
+      {
+        bool: {
+          must: [
+            { term: { category: "improving" } }
+          ]
+        }
+      }
+    ]
+    if window
+      filter.unshift(
+        {
+          range: {
+            created_at: {
+              gte: window
+            }
+          }
+        }
       )
-      break if new_values.empty?
-
-      improving_identifiers.concat( new_values )
-      frequency_hash = improving_identifiers.group_by( &:itself ).transform_values( &:count )
-      sorted_frequency_hash = frequency_hash.sort_by {| _, count | -count }.to_h
-      enough_candidates = sorted_frequency_hash.count > validator_redundancy_factor &&
-        sorted_frequency_hash.values[validator_redundancy_factor - 1] >= improving_id_threshold
-      page += 1
     end
+    filter.last[:bool][:must] << { term: { "taxon.id": taxon_id } } if taxon_id
+    if place_id
+      filter.last[:bool][:must] << {
+        terms: { "observation.place_ids": Array( place_id ) }
+      }
+    end
+    aggregation = {
+      user_id: {
+        terms: {
+          field: "user.id",
+          size: 1000
+        }
+      }
+    }
+    begin
+      response = Identification.elastic_search(
```

Comment:
The goal is to get users who have made >=improving_id_threshold improving IDs for a taxon_id (and optionally within time and place constraints).
the [/observations/identifiers](https://api.inaturalist.org/v1/docs/#!/Observations/get_observations_identifiers) API endpoint can't be used because it is not constrained to improving IDs
Before I was paginating through [/identifications](https://api.inaturalist.org/v1/docs/#!/Identifications/get_identifications) API endpoint which to my knowledge lacks an aggregation by user_id
this is why the ES query is much faster, because it is constrained to improving IDs but includes an aggregation by user_id

Reply:
There is the `/v1/identifications/identifiers` endpoint which has parameters `d1`, `category`, `taxon_id`, and `place_id`. It does only return a maximum of 500 users not the 1000 you're requesting here, but we could find a way to improve that if that would otherwise work for you